.buildkite/nightly_steps.yml

```diff
@@ -2,133 +2,290 @@ env:
   DOCKERFILE_FTEST_PATH: "Dockerfile.ftest.wolfi"
 
 steps:
-  - label: "🔨 MySQL"
-    command:
-      - ".buildkite/run_nigthly.sh mysql"
+  - label: "🔨 [Python {{ matrix }}] MySQL"
+    command: ".buildkite/run_functional_test.sh"
+    env:
+      PYTHON_VERSION: "{{ matrix }}"
+      CONNECTOR: "mysql"
+      DATA_SIZE: "small"
     artifact_paths:
       - "perf8-report-*/**/*"
-  - label: "🔨 Network Drive"
-    command:
-      - ".buildkite/run_nigthly.sh network_drive"
+    matrix:
```

Comment:
TIL about "matrix"
would it be worth it to just redo this whole file as a single step, using `matrix.setup` to define python version, connector service type, and data size? Looking at https://buildkite.com/docs/pipelines/build-matrix

Reply:
Yup good idea!

Reply:
Ah, service type being > 20? Boo. 
Well thanks for looking. :sigh:

Reply:
We can actually make it work too, just breaking into matrices by 20 service types x 2 versions, or 3 versions x 16 service types too, but can do later :)